src/scatter_buildblock/ScatterEstimation.cxx

```diff
@@ -0,0 +1,1486 @@
+                                                                    this->input_projdata_sptr->get_proj_data_info_sptr()->create_shared_clone(),
+                                                                    out_filename,
+                                                                    std::ios::in | std::ios::out | std::ios::trunc));
+            this->back_projdata_sptr->fill(0.0f);
+        }
+    }
+
+
+
+    if (run_in_2d_projdata)
+    {
+        // Normalise in order to get the additive component
+
+        std::string out_filename = extras_path.get_path() +"/"+ output_background_estimate_prefix + "_0_2d.hs";
+        add_projdata_2d_sptr.reset(
+                    new ProjDataInterfile(this->input_projdata_2d_sptr->get_exam_info_sptr(),
+                                          this->input_projdata_2d_sptr->get_proj_data_info_sptr() ,
+                                          out_filename,
+                                          std::ios::in | std::ios::out | std::ios::trunc));
+        add_projdata_2d_sptr->fill(*back_projdata_2d_sptr);
+        this->multiplicative_binnorm_2d_sptr->apply(*this->add_projdata_2d_sptr, start_time, end_time);
+
+        iterative_object->get_objective_function_sptr()->set_additive_proj_data_sptr(this->add_projdata_2d_sptr);
+    }
+    else
+    {
+        // Normalise in order to get the additive component
+        std::string out_filename = extras_path.get_path() +"/"+ output_background_estimate_prefix + "_0.hs";
+        add_projdata_2d_sptr.reset(
+                    new ProjDataInterfile(this->input_projdata_sptr->get_exam_info_sptr(),
+                                          this->input_projdata_sptr->get_proj_data_info_sptr() ,
+                                          out_filename,
+                                          std::ios::in | std::ios::out | std::ios::trunc));
+        add_projdata_sptr->fill(*back_projdata_sptr);
+        this->multiplicative_binnorm_sptr->apply(*this->add_projdata_sptr, start_time, end_time);
+
+        iterative_object->get_objective_function_sptr()->set_additive_proj_data_sptr(this->add_projdata_sptr);
+    }
+
+    return Succeeded::yes;
+}
+
+Succeeded
+ScatterEstimation::
+set_up_analytic()
+{
+    //TODO : I have most stuff in tmp.
+     error("Analytic recon not implemented yet");
+    return Succeeded::yes;
+}
+
+Succeeded
+ScatterEstimation::
+process_data()
+{
+
+    if (this->set_up() == Succeeded::no)
+    {
+        info("ScatterEstimation: Unsuccessful set up!");
+       return Succeeded::no;
+    }
+
+    const double start_time = this->input_projdata_sptr->get_exam_info_sptr()->get_time_frame_definitions().get_start_time();
+    const double end_time =this->input_projdata_sptr->get_exam_info_sptr()->get_time_frame_definitions().get_end_time();
+
+    float local_min_scale_value = 0.5f;
+    float local_max_scale_value = 0.5f;
```

Comment:
much better to move these declarations to where they are actually used.

Reply:
Look at line 1045, I replace their values with the values of `this->max_scale_value`. when the scatter iteration is larger than 1. 
This way do not replicate code.